au4/src/projectscene/qml/Audacity/ProjectScene/clipsview/Timeline.qml

```diff
@@ -42,13 +42,14 @@ Rectangle {
     }
     // ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    StyledTextLabel {
+    TimelineRuler {
+        id: ruler
+        offset: timelineContext.offset
+        zoom: timelineContext.zoom
```

Comment:
It may be better to pass the entire context as a property, this is done in other places, this can be useful when more properties from it are needed (for example, a selected loop)